### ISSUE TITLE
Create a new category

### ISSUE THREAD

**Issue by mohammad425** [pressidium/pressidium-cookie-consent#13]
How to create a new category? And why can't a cookie be added to the "necessary" category?

**Comment by over-engineer** [comment]
Hey, @mohammad425!

>How to create a new category?

Currently, there’s no way to create a new cookie category. Our goal was to maintain a balance between flexibility and ease of use.

That being said, we recognize that unique use cases might require new categories. Could you please share any specific use case or examples of new categories you would like to create?

>why can't a cookie be added to the "necessary" category?

Adding cookies to a category serves two purposes:

- Deleting them when a user opts-out of that cookie category inside cookie settings (when the “Auto-clear cookies” option is enabled, under the “General” tab)

- Listing them in the cookie table under that cookie category inside cookie settings

Deleting necessary cookies is not a thing, since those cookies are essential for the proper functioning of the website and they do _not_ require the explicit consent of a user (i.e. users are not able to opt-out of the “Necessary” cookie category)

However, you should be able to list necessary cookies to inform you users about them.

I’ve opened a separate issue to track the ability to list necessary cookies at #20, since this one is titled “Create a new category” so it’s mainly about your first question.

**Comment by mohammad425** [comment]
Users should be able to categorize their cookies precisely. In my opinion, it would be great if users could add their own categories. However, if this option is not available, the following categories should be added:

| Category      | Description                                                                                                                         |
|---------------|-------------------------------------------------------------------------------------------------------------------------------------|
| Performance   | Performance cookies monitor site performance and track user actions, but they do not collect identifiable information. They collect data anonymously and use it to improve the website. Performance cookies can count page visits, analyze the time users spend on a website, and examine loading speeds to enhance performance. They can be first-party or third-party cookies.                                                |
| External Media| Content from video platforms and social media platforms is typically blocked by default. If cookies from external media are accepted, accessing this content will no longer require manual consent.                                                                              |
| Functionality | Functional cookies primarily enhance the performance of a website, as certain functions may not be available without them. They allow the website to remember user preferences and settings, e.g., user location to display personalized news stories and weather reports. Consent is required for this to happen.                   |
| Marketing     | Used for targeting users with personalized ads. For instance, the Facebook pixel can be used to show specific ads to certain users.                                                                   |
| Unclassified  | Unclassified cookies do not belong to any other category or are currently in the process of categorization (e.g., m.m.stripe.com).                                                                   |

However, it is essential to consider that not all users may need all of these categories. Therefore, if a cookie is not set for a specific category, it should not be displayed on the front end (for visitors) to avoid table clutter.

**Comment by over-engineer** [comment]
@mohammad425 Thanks for bringing up these valid points. Let’s leave this open, and in a future iteration, we’ll work on allowing users to add their own custom cookie categories.

**Comment by formaciongrafica** [comment]
Excellent plugin! Adding "standard" categories or allowing customization would make it an even better solution!
Thank you!
